### PR TITLE
fix: Make execution state use auth crate

### PR DESCRIPTION
**Pull request by Zertsov**
### Description
How about I do a PR this time?
This updates the execution state to use the `turborepo_auth` crate, beginning the end of the `CommandBase` as the token holder.

### Testing Instructions
Run a command with the `--go-fallback`. This should pass.

Closes TURBO-1987

### CODE REVIEW

File: crates/turborepo-auth/src/lib.rs [hunk 1]
```diff
@@ -112,7 +113,7 @@ mod tests {
 
         let client = MockApiClient::new();
```
Comment (chris-olszewski) [inline crates/turborepo-auth/src/lib.rs:114]: For future PR, but we can get rid of the mock client usage in this test now.

File: crates/turborepo-lib/src/execution_state.rs [hunk 2]
```diff
@@ -49,8 +49,22 @@ impl<'a> TryFrom<&'a CommandBase> for ExecutionState<'a> {
 
         let config = base.config()?;
 
+        // TODO(voz): Don't unwrap here. Should be fine since this can rarely ever be
+        // None.
+        let auth_file_path = base.global_auth_path.clone().unwrap();
+        let config_file_path = base.global_config_path.clone().unwrap();
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/execution_state.rs:55]: I'm pretty sure you want 
```suggestion
        let auth_file_path = base.global_auth_path().unwrap();
        let config_file_path = base.global_config_path().unwrap();
```
as the fields themselves will always be `None` except for in tests.

We also can't be unwrapping like this and should instead go through normal error handling if there's a config error.

File: crates/turborepo-lib/src/commands/mod.rs [hunk 3]
```diff
@@ -30,9 +30,7 @@ pub(crate) mod unlink;
 pub struct CommandBase {
     pub repo_root: AbsoluteSystemPathBuf,
     pub ui: UI,
-    #[cfg(test)]
     pub global_config_path: Option<AbsoluteSystemPathBuf>,
-    #[cfg(test)]
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/commands/mod.rs:35]: I don't understand why we're removing the `#[cfg(test)]` here. They should be added back

**Review by chris-olszewski — CHANGES_REQUESTED** [review]
Concerned about the making of test only fields now available in prod code.

File: crates/turborepo-lib/src/execution_state.rs (hunk 2 above)
Comment (Zertsov) [reply crates/turborepo-lib/src/execution_state.rs:55]: I was doing the `clone` due to move issues, but I can just make those methods public instead.

I'll handle unwraps

File: crates/turborepo-lib/src/commands/mod.rs (hunk 3 above)
Comment (Zertsov) [reply crates/turborepo-lib/src/commands/mod.rs:35]: I can make the getters public, but this was to allow for the execution state to know where the auth crate should look for tokens

File: crates/turborepo-lib/src/execution_state.rs [hunk 4]
```diff
@@ -49,8 +49,22 @@ impl<'a> TryFrom<&'a CommandBase> for ExecutionState<'a> {
 
         let config = base.config()?;
 
+        // TODO(voz): Don't unwrap here. Should be fine since this can rarely ever be
```
Comment (Zertsov) [inline crates/turborepo-lib/src/execution_state.rs:52]: I'll remove this after the Github Actions finish - don't wanna restart them now

**Review by chris-olszewski — APPROVED** [review]
LGTM

**Pull request merged**